src/app/MessageDef/CommandPath.cpp

```diff
@@ -30,12 +30,48 @@
 #include <stdio.h>
 
 #include <app/AppBuildConfig.h>
+#include <app/data-model/Decode.h>
+#include <app/data-model/Encode.h>
 
 using namespace chip;
 using namespace chip::TLV;
 
 namespace chip {
 namespace app {
+
+namespace CommandPath {
+CHIP_ERROR Type::Decode(chip::TLV::TLVReader & reader)
+{
+    CHIP_ERROR err;
+    TLV::TLVType outer;
+
+    err = reader.EnterContainer(outer);
+    ReturnErrorOnFailure(err);
+
+    while ((err = reader.Next()) == CHIP_NO_ERROR)
```

Comment:
Even if you were to write the code in a way that expects that order, it'll still end up looking like this, because of the fact that these tags _may_ not be present.

At this point, adding code to enforce ordering will be extra and only serves to make this less flexible.

Reply:
You could in fact write code that enforces ordering and handles missing tags without too much trouble.

The real question is this: is this code spec-compliant?  The spec requires aborting interactions on malformed messages.  Is a message with out-or-order tags malformed?  If not, the spec needs to be _really_ clear about that.